synapse/cores/ram.py

```diff
@@ -64,8 +64,8 @@ def _delRowsByIdProp(self, iden, prop):
     def _getRowsByIdProp(self, iden, prop):
         return [ row for row in self.rowsbyid.get(iden) if row[1] == prop ]
 
-    def _delRowsByProp(self, prop, valu=None, mintime=None, maxtime=None):
-        for row in self.getRowsByProp(prop,valu=valu,mintime=mintime,maxtime=maxtime):
+    def _delRowsByProp(self, prop, valu=None, mintime=None, maxtime=None, timeout=None):
```

Comment:
timeout options are probably only appropriate for "getters", as it creates a nasty problem for storage layers (other than postgres ) to implement "rollback" behavior an we wouldn't want the timeout to cause a partial delete